content/src/service/caching/Cache.ts

```diff
@@ -0,0 +1,77 @@
+import { EntityType } from 'dcl-catalyst-commons'
+import LRU from 'lru-cache'
+
+/** This is a regular cache, that can be configured with a max size of elements */
+export class Cache<Key, Value> {
+  // Null implies that there is no value set for the key even though one was calculated for it.
+  // The cache lib returns undefined when no value was calculated at all.
+  private internalCache: LRU<Key, Value | null>
+
+  constructor(size: number) {
+    this.internalCache = new LRU({ max: size })
+  }
+
+  /**
+   * Get the values for the given keys, or calculated them if they are nor present.
+   * If no value can be calculated for a given key, then the key shouldn't be present, or it should be, with an `undefined` value.
+   */
+  async get(keys: Key[], orCalculate: (keys: Key[]) => Promise<Map<Key, Value | undefined>>): Promise<Value[]> {
+    // Check what is on the cache
+    const onCache: Map<Key, Value | null | undefined> = new Map(keys.map((key) => [key, this.internalCache.get(key)]))
+
+    // Get all the keys without values calculated
+    const missing: Key[] = Array.from(onCache.entries())
+      .filter(([, value]) => value === undefined)
+      .map(([key]) => key)
+
+    // Calculate values for those keys
+    const calculated: Map<Key, Value | undefined> = missing.length > 0 ? await orCalculate(missing) : new Map()
+
+    // Save the calculated values. When a key doesn't have a value, set it as null
+    Array.from(calculated.entries())
+      .map<[Key, Value | null]>(([key, value]) => [key, value ?? null])
+      .forEach(([key, value]) => this.internalCache.set(key, value))
```

Comment:
`for(const [key, value] of calculated)` is streamable (every iteration consumes the iterator) and memory efficient in comparison to this currying approach